src/operators/select/select.cc

```diff
@@ -130,26 +131,28 @@ arrow::Datum Select::Filter(const std::shared_ptr<Block> &block,
       uint8_t val = std::static_pointer_cast<arrow::UInt8Scalar>(
                         predicate->value_.scalar())
                         ->value;
+      auto datum_val = predicate->value_;
 
       switch (predicate->comparator_) {
```

Comment:
If we add more types (i.e. int32_t) would we need to copy the switch statement for the predicate type? can we make that statement into a separate function?

Reply:
We may want to avoid the type switch statement entirely through a template-based approach. In general, we should probably save this kind of change for a general overhaul to the select logic. Added issue #83.